Core/FrameFilter.cpp

```diff
@@ -25,32 +25,14 @@ FrameFilter::FrameFilter(u8 gender, u8 ability, u8 shiny, bool skip, const QVect
 {
 }
 
-bool FrameFilter::compareFrame(const Frame &frame) const
+bool FrameFilter::compareShiny(const Frame &frame) const
 {
-    if (skip)
-    {
-        return true;
-    }
-
-    if (!natures.at(frame.getNature()))
-    {
-        return false;
-    }
-
-    if (gender != 255 && gender != frame.getGender())
-    {
-        return false;
-    }
-
-    if (ability != 255 && ability != frame.getAbility())
-    {
-        return false;
-    }
+    return skip || shiny == 255 || (shiny & frame.getShiny());
```

Comment:
Is this functionally the same as the old check?

Reply:
If I'm not mistaken, `frame.getShiny()` returns 0, 1 or 2. `shiny`'s possible values are  255, 1, 2 or 3.  `compareShiny()` should return `true` if and only if `skip == true || shiny == 255 || shiny == frame.getShiny() || (shiny == 3 && (frame.getShiny() == 1 || frame.getShiny() == 2)) `

If `skip == true || shiny == 255` fails, which means `shiny != 255` , the true table will be like:

<table>
    <tr>
        <td colspan="2", rowspan="2"></td>
        <td colspan="3"><code>frame.getShiny()</code></td>
    </tr>
    <tr>
        <td>0</td>
        <td>1</td>
        <td>2</td>
    </tr>
    <tr>
        <td rowspan="3"><code>shiny</code></td>
        <td>1</td>
        <td></td>
        <td><code>true</code></td>
        <td></td>
    </tr>
    <tr>
        <td>2</td>
        <td></td>
        <td></td>
        <td><code>true</code></td>
    </tr>
    <tr>
        <td>3</td>
        <td></td>
        <td><code>true</code></td>
        <td><code>true</code></td>
    </tr>
</table>

So it's same as `(shiny & frame.getShiny())`.